app/models/manager_refresh/inventory_collection_default/infra_manager.rb

```diff
@@ -0,0 +1,111 @@
+class ManagerRefresh::InventoryCollectionDefault::InfraManager < ManagerRefresh::InventoryCollectionDefault
+  class << self
+    def vms(extra_attributes = {})
+      attributes = {
+        :model_class => ::ManageIQ::Providers::InfraManager::Vm,
+        :association => :vms,
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def disks(extra_attributes = {})
+      attributes = {
+        :model_class => ::Disk,
+        :manager_ref => [:hardware, :device_name],
+        :association => :disks
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def nics(extra_attributes = {})
+      attributes = {
+        :model_class => ::Network,
+        :manager_ref => [:hardware, :ipaddress],
+        :association => :nics
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def hardwares(extra_attributes = {})
+      attributes = {
+        :model_class => ::Hardware,
+        :manager_ref => [:vm_or_template],
+        :association => :hardwares
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def datacenters(extra_attributes = {})
+      attributes = {
+        :model_class          => ::Datacenter,
+        # TODO: seems not to be correct
+        :association          => :datacenters,
+        :attributes_blacklist => [:ems_children]
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def resource_pools(extra_attributes = {})
+      attributes = {
+        :model_class          => ::ResourcePool,
+        :association          => :resource_pools,
+        :attributes_blacklist => [:ems_children]
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def clusters(extra_attributes = {})
+      attributes = {
```

Comment:
in the parser you use uid_ems as manager ref, so in this case you should have here

:manager_ref => [:uid_ems]